lib/Parser.js

```diff
@@ -624,10 +624,16 @@ class Parser extends Tapable {
 	}
 
 	walkMethodDefinition(methodDefinition) {
+		// TODO: Use `inScope`
```

Comment:
Actually it's not really required. When the parser visits `methodDefinition.value` it calls `inScope` which will change `topLevelScope` because the value is a `FunctionDeclaration`. The tricky case is when class properties will land in the spec because of arrow functions.

**tl;dr**: Currently it's not required to change `topLevelScope` here but it may be mandatory to properly handle the future syntax. Maybe this should be moved to the class definition visitor.

What's your thoughts on this?

Reply:
Ah I see, then remove all changes in walkMethodDefinition.

ClassProperty will probably a separate AST node type next to MethodDefinition.

Maybe like this: https://astexplorer.net/#/gist/732b089f326bb5ad9260decf2656b7c0/eb3cb03930770076e08d514887930b7960309d24